pkg/util/cryptomaterial/certchains/certchains.go

```diff
@@ -48,6 +48,14 @@ func (cs *CertificateChains) GetCertKey(certPath ...string) ([]byte, []byte, err
 	return signer.GetCertKey(certPath[len(certPath)-1])
 }
 
+func (cs *CertificateChains) Regenerate(certPath ...string) error {
+	if signer := cs.GetSigner(certPath[0]); signer != nil {
+		return signer.Regenerate(certPath[1:]...)
+	}
+
+	return fmt.Errorf("no such signer: %s", certPath[0])
```

Comment:
shouldn't be using klog instead of fmt?

Reply:
I think fmt here is fine; it is not printing it to the console but building an error to return.

Reply:
True! my bad, I misread it by printf xD